alpha/apps/kaltura/lib/storage/urlTokenizers/kWowzaSecureTokenUrlTokenizer.php

```diff
@@ -0,0 +1,157 @@
+<?php
+
+class kWowzaSecureTokenUrlTokenizer extends kUrlTokenizer
+{
+	
+	/**
+	 * @var string
+	 */
+	protected $paramPrefix;
+	
+	/**
+	 * @var bool
+	 */
+	protected $shouldIncludeClientIp;
+	
+	/**
+	 * @var string
+	 */
+	protected $hashAlgorithm;
+
+	
+	public function tokenizeSingleUrl($url, $urlPrefix = null)
+	{
+		$tokenPrefix = $this->getParamPrefix();
+		$expiryTimeFrame = $this->getWindow();
+		
+		$params = array();
+		
+		$params["starttime"] = time();
+		$params["endtime"] = time() + $expiryTimeFrame;
+		
+		$prefixedParams = $this->prefixParams($tokenPrefix,$params);
+		
+		return $url."?".http_build_query($prefixedParams,'','&').'&'.$tokenPrefix.'hash='.$this->getHash($prefixedParams,$url);
```

Comment:
IMHO it would be a bit cleaner to do $prefixedParams[$tokenPrefix.'hash'] = $this->getHash($prefixedParams,$url); before the call to http_build_query